### DISCUSSION TITLE
How do you execute "git add ..." from Neo-tree filesystem window view?

### DISCUSSION THREAD

**Discussion by stephane-klein** [nvim-neo-tree/neo-tree.nvim#528] [Q&A] [answered]
In the bottom window in this screenshot you can see that `/scripts/init.sh` isn't tracked by git:

![Capture d’écran du 2022-09-18 13-00-53](https://user-images.githubusercontent.com/54498080/190898784-e51576cb-3fd6-4f82-9787-831ff35111d3.png)

I would like to execute `git add scripts/init.sh` via Neo-tree window.

When I press `?` in Neo-tree window, I see that:

![Capture d’écran du 2022-09-18 13-04-43](https://user-images.githubusercontent.com/54498080/190898924-9d9a24c0-bfb6-4595-9afc-d6c9c052461a.png)

I see nothing to perform a `git add...` command.

From [this documentation](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/35e9f9b53d05bfec90314865787cd920a8b468aa/README.md?plain=1#L306), I tried to add this lines in my Neo-tree setup config:

```
                window = {
                    mappings = {
                        ["<space>"] = {
                            "toggle_node",
                            nowait = false, -- disable `nowait` if you have existing combos starting with this char that you want to use
                        }--,
                        ["A"]  = "git_add_all",
                        ["gu"] = "git_unstage_file",
                        ["ga"] = "git_add_file",
                        ["gr"] = "git_revert_file",
                        ["gc"] = "git_commit",
                        ["gp"] = "git_push",
                        ["gg"] = "git_commit_and_push"
                    }
                },
```

<details>
<summary>Click to see complete Neo-tree config</summary>

```
    use {
        "nvim-neo-tree/neo-tree.nvim",
        branch = "v2.x",
        config = function ()
            -- Unless you are still migrating, remove the deprecated commands from v1.x
            vim.cmd([[ let g:neo_tree_remove_legacy_commands = 1 ]])
            require("neo-tree").setup({
                close_if_last_window = false, -- Close Neo-tree if it is the last window left in the tab
                open_files_in_last_window = false,
                popup_border_style = "rounded",
                enable_git_status = true,
                enable_diagnostics = true,
                window = {
                    mappings = {
                        ["<space>"] = {
                            "toggle_node",
                            nowait = false, -- disable `nowait` if you have existing combos starting with this char that you want to use
                        }--,
                        ["A"]  = "git_add_all",
                        ["gu"] = "git_unstage_file",
                        ["ga"] = "git_add_file",
                        ["gr"] = "git_revert_file",
                        ["gc"] = "git_commit",
                        ["gp"] = "git_push",
                        ["gg"] = "git_commit_and_push"
                    }
                },
                filesystem = {
                    hijack_netrw_behavior = "open_current"
                }
            })
        end
    }
```
</details>

But that doesn't work. I have an error when I press `?` in Neo-tree window.

Question: how do you execute "git add ..." from Neo-tree filesystem window view?

Best regards,  
Stéphane

**Comment by cseickel** [comment, marked as answer]
As I responded on reddit:

> There is no built-in command to do that right now, but it is a fairly trivial command that you can create by mapping to a function. I will also consider adding the git commands to the "common" commands that are shared by all sources.

I just looked at the code and there is no reason why the git commands can't be moved from the "git_status" source to "common" where they will be accessible from the "filesystem" source. I'll make an issue from this discussion and put it on the todo list for the next release.

**Reply by cseickel** [reply]
This has just been released. I did not add the mappings to other sources by default, but if you add them manually like you did in your example above then it will work now.

**Reply by stephane-klein** [reply]
@cseickel thanks, it works perfectly :+1:
